tests/flow/test_call_subquery.py

```diff
@@ -42,30 +42,104 @@ def expect_error(self, query, expected_err_msg):
 
-        import_error = "WITH imports in CALL {} must consist of only simple references to outside variables"
-        match_after_updating = "A WITH clause is required to introduce MATCH after an updating clause"
-        union_column_name_error = "All sub queries in a UNION must have the same column names."
-        queries_errors = [
-            ("WITH 1 AS a CALL {WITH a+1 AS b RETURN b} RETURN b", import_error),
-            ("WITH {a: 1} AS map CALL {WITH map.a AS b RETURN b} RETURN b", import_error),
-            ("WITH [1, 2, 3] AS list CALL {WITH list[0] AS b RETURN b} RETURN b", import_error),
-            ("WITH 'RAZ' AS str CALL {WITH toUpper(str) AS b RETURN b} RETURN b", import_error),
-            ("WITH 1 AS a CALL {WITH a AS b RETURN b} RETURN b", import_error),
-            ("WITH 1 AS a CALL {WITH a LIMIT 5 RETURN a} RETURN a", import_error),
-            ("WITH 1 AS a CALL {WITH a ORDER BY a.v RETURN a} RETURN a", import_error),
-            ("WITH 1 AS a CALL {WITH a WHERE a > 5 RETURN a} RETURN a", import_error),
-            ("WITH 1 AS a CALL {WITH a SKIP 5 RETURN a} RETURN a", import_error),
-            ("WITH true AS a CALL {WITH NOT(a) AS b RETURN b} RETURN b", import_error),
-            ("CALL {CREATE (n:N) MATCH (n:N) RETURN n} RETURN 1", match_after_updating),
-            ("WITH 1 AS a CALL {WITH a CREATE (n:N) MATCH (n:N) RETURN n} RETURN a", match_after_updating),
-            ("CALL {MATCH (n:N) CREATE (n:N2)} RETURN 1 ", "The bound variable 'n' can't be redeclared in a CREATE clause"),
-            ("MATCH (n) CALL {WITH n AS n1 RETURN n1 UNION WITH n RETURN n1} RETURN n, n1", import_error),
-            ("MATCH (n) CALL {WITH n RETURN n AS n1 UNION WITH n AS n1 RETURN n1} RETURN n, n1", import_error),
-            ("CALL {RETURN 1 AS one UNION RETURN 2 AS two} RETURN 1", union_column_name_error),
-            ("MATCH (n) CALL {RETURN 1 AS one UNION RETURN 2 AS two} RETURN 1", union_column_name_error)
+        # invalid queries: imports that are not simple imports
+        queries = [
+            "WITH 1 AS a CALL {WITH a + 1 AS b RETURN b} RETURN b",
+            "WITH {a: 1} AS map CALL {WITH map.a AS b RETURN b} RETURN b",
+            "WITH [1, 2, 3] AS list CALL {WITH list[0] AS b RETURN b} RETURN b",
+            "WITH 'RAZ' AS str CALL {WITH toUpper(str) AS b RETURN b} RETURN b",
+            "WITH 1 AS a CALL {WITH a AS b RETURN b} RETURN b",
+            "WITH 1 AS a CALL {WITH a LIMIT 5 RETURN a} RETURN a",
+            "WITH 1 AS a CALL {WITH a ORDER BY a.v RETURN a} RETURN a",
+            "WITH 1 AS a CALL {WITH a WHERE a > 5 RETURN a} RETURN a",
+            "WITH 1 AS a CALL {WITH a SKIP 5 RETURN a} RETURN a",
+            "WITH true AS a CALL {WITH NOT(a) AS b RETURN b} RETURN b",
+            "MATCH (n) CALL {WITH n AS n1 RETURN n1 UNION WITH n RETURN n1} RETURN n, n1",
+            "MATCH (n) CALL {WITH n RETURN n AS n1 UNION WITH n AS n1 RETURN n1} RETURN n, n1",
```

Comment:
what's the issue with this last query ?

Reply:
Non-simple import in second branch of the `UNION`.

Reply:
Non-simple import in the second branch of the `UNION`.